src/custom-hooks/useGetData.js

```diff
@@ -1,28 +1,28 @@
-import { collection, getDocs, onSnapshot } from "firebase/firestore"
-import { useEffect, useState } from "react"
+import { collection, onSnapshot } from "firebase/firestore"
+import { useCallback, useEffect, useMemo, useState } from "react"
 import { db } from "../firebase.config"
 
 const useGetData = (collectionName) => {
   const [data, setData] = useState([])
   const [loading, setLoading] = useState(true)
-  const collectionRef = collection(db, collectionName)
+  const collectionRef = useMemo(
+    () => collection(db, collectionName),
+    [collectionName]
+  )
+
+  const getData = useCallback(async () => {
+    // ==== firebase firestore realtime data update =====
+    onSnapshot(collectionRef, (snapshot) => {
+      const { docs } = snapshot
+      const mappedDocs = docs.map((doc) => ({ ...doc.data(), id: doc.id }))
+      return setData(mappedDocs)
+    })
+  }, [collectionRef])
```

Comment:
1. Wonderful work my mentor. I appreciate your time. I noticed getData is made to be asycn function, but I could not see `await` anywhere within the function. I want to know why that is so. Thanks.

Reply:
I think I forgot to remove it, so please do so!